perf/istio-install/create_cluster.sh

```diff
@@ -261,18 +264,41 @@ data:
   trustDomain: ${PROJECT_ID}.svc.id.goog
   gkeClusterUrl: https://container.googleapis.com/v1/projects/${PROJECT_ID}/locations/${ZONE}/clusters/${CLUSTER_NAME}
 EOF
+)
 
-export KUBECONFIG="${CLUSTER_NAME}/kube.yaml"
+export KUBECONFIG="${WD}/tmp/${CLUSTER_NAME}/kube.yaml"
 gcloud container clusters get-credentials "${CLUSTER_NAME}" --zone "${ZONE}"
 
-kubectl create clusterrolebinding cluster-admin-binding \
-  --clusterrole=cluster-admin \
-  --user="$(gcloud config get-value core/account)"
+# The gcloud key create command requires you dump its service account
+# credentials to a file. Let that happen, then pull the contents into a varaible
+# and delete the file.
+CLOUDKEY=""
+if [[ "${CLUSTER_NAME}" != "" ]]; then 
+  if ! kubectl -n kube-system get secret google-cloud-key >/dev/null 2>&1 || ! kubectl -n istio-system get secret google-cloud-key > /dev/null 2>&1; then
+    gcloud iam service-accounts keys create "${WD}/tmp/${CLUSTER_NAME}/${CLUSTER_NAME}-cloudkey.json" --iam-account="${GCP_CTL_SA}"@"${PROJECT_ID}".iam.gserviceaccount.com
+    # Read from the named pipe into the CLOUDKEY variable
+    CLOUDKEY=$(cat "${WD}/tmp/${CLUSTER_NAME}/${CLUSTER_NAME}-cloudkey.json")
+    # Clean up
+    rm "${WD}/tmp/${CLUSTER_NAME}/${CLUSTER_NAME}-cloudkey.json"
+  fi
+fi
+
+if ! kubectl get clusterrolebinding cluster-admin-binding > /dev/null 2>&1; then 
+  kubectl create clusterrolebinding cluster-admin-binding \
+    --clusterrole=cluster-admin \
+    --user="$(gcloud config get-value core/account)"
+fi
 
 # Update the cluster with the GCP-specific configmaps
-kubectl -n kube-system apply -f "${CLUSTER_NAME}/configmap-neg.yaml"
-kubectl -n kube-system create secret generic google-cloud-key  --from-file key.json="${CLUSTER_NAME}/google-cloud-key.json"
+if ! kubectl -n kube-system get secret google-cloud-key > /dev/null 2>&1; then 
+  kubectl -n kube-system create secret generic google-cloud-key  --from-file key.json=<(echo "${CLOUDKEY}")
```

Comment:
does create fail if it already exists? If it doesn't it seems reasonable to do this always

Reply:
Another part of the code conditionally generates the google-cloud-key only if the secret is missing from one of the two namespaces. If we don't have to create another service account key, we shouldn't. I suppose the condition could be changed to check if the CLOUDKEY variable is empty instead, if you would prefer.